### DISCUSSION TITLE
CMS Validator Use Case

### DISCUSSION THREAD

**Discussion by rahulpkomodo** [CMSgov/price-transparency-guide-validator#64] [Q&A] [answered]
Hey! Thanks for building this nifty file validator. 

In terms of users of this repo, is this for health plans to validate their json files before posting on their websites or for third-party users to validate what health plans have posted on their website or both? Essentially, trying to figure out if data posted on a health plans website should already fit the schema. This may not be the forum to ask this question, but let me know where to look! 

Thanks!

**Comment by olderthandirt5** [comment, marked as answer]
Just another user, so take this with a grain of salt. It is being used by both sets of users, in my experience. We use it to validate files before we post them, and I know some of our partners use it to validate the files. This eliminates a lot of confusion and gives all sides a "single source of truth" if there is a disagreement. As a developer, it is invaluable!

**Comment by shaselton-usds** [comment]
@rahulpkomodo Sorry for the delayed response. It can be used by anyone. I would expect the most use-cases to be for those developing the files to determine if their files are "valid" in terms of JSON schema structure (so, probably a good step to throw into whatever development pipeline that is established) and to be used by those consuming the files to ensure that their scripts will not break based off of shared understandings on expectations of the file's structure. 

@olderthandirt5 is basically hitting the nail on the head here.

**Comment by bolthasenteredchat** [comment]
just came upon this.  guys, this is an absolute gamechanger for the industry.  if you guys have seen the hl7fhir.frmwk this is a very large part of what the guys over there had not gotten to yet.  take these .mrf's, and I mean all of them, frame them out to where you can very accurately see 100% of the entire spend index, and if you can do that, you could very possibly bridge that platform with fhir.

now picture the solution this provides on a highly segmented $4T  (20% GDP) market.  a market so complex, so containerized, the smartest guys in the room don't give you actual numbers until 365 days pass.

can you imagine the information you'd have instant access to.....can you imagine the light speed velocity of how medicine would be completely revolutionized.  Where AI would literally transform the entire world. Can you imagine the transformation humankind would experience? Can you imagine the cures we would find due to the entirety of knowledge and information we would have?

if you see what all of that potential looks like, suddenly, going to the moon just feels like we've been there before.
